Source/WebCore/css/quirks.css

```diff
@@ -23,6 +23,21 @@
 
 @namespace "http://www.w3.org/1999/xhtml";
 
+/* Force inside position for orphaned lis */
+li {
+    list-style-position: inside;
+}
+
+/* Restore outside position for lists inside lis */
+li :is(ul, ol) {
+    list-style-position: outside;
+}
+
+/* Undo previous two rules for properly nested lists */
+:is(ul, ol) :is(ul, ol, li) {
+    list-style-position: unset;
+}
```

Comment:
@Emilio I saw gecko's quirks stylesheet also includes `dir` and `menu`. However, WebKit and Blink have both shipped this quirk without these 2 selectors, so I think it's probably safe for Gecko to remove those 2 selectors as well.

Reply:
Yeah, fair. Thanks!